eslint not working - errors in logs

### Summary

<!-- Begin your issue with a one sentence summary -->
Eslint is not working in Zed, however it works in the terminal.

### Description
<!--  Describe with sufficient detail to reproduce from a clean Zed install.
  - Any code must be sufficient to reproduce (include context!)
  - Code must as text, not just as a screenshot.
  - Issues with insufficient detail may be summarily closed.
-->

Zed:
![Image](https://github.com/user-attachments/assets/a88faf85-fc42-4620-a87c-1ff010fed933)

Terminal:
![Image](https://github.com/user-attachments/assets/39dc886d-e58c-4cf2-b88f-b400910232c6)

Steps to reproduce:
1. open file with lint error and see no error shown in Zed

**Expected Behavior**:
eslint errors showing in Zed.

**Actual Behavior**:
Only errors showing in terminal.

<!-- Before Submitting, did you:
  1. Include settings.json, keymap.json, .editorconfig if relevant?
  2. Check your Zed.log for relevant errors? (please include!)
  3. Click Preview to ensure everything looks right?
  4. Hide videos, large images and logs in ``` inside collapsible blocks:

<details><summary>click to expand</summary>

```json

```
</details>
-->
Logs:
<details><summary>click to expand</summary>

```bash
2025-06-27T15:27:59-06:00 WARN  [project::lsp_store] unhandled capability registration: Registration { id: "7eba4d7b-7806-4f1d-830e-be83a5ba6a91", method: "workspace/didChangeWorkspaceFolders", register_options: Some(Object {}) }
2025-06-27T15:28:01-06:00 INFO  [lsp] Language server with id 10 sent unhandled notification eslint/status:
{
  "uri": "file:///home/mk/Desktop/CODE/amboss-auth/src/app/actions.ts",
  "state": 1,
  "validationTime": 1900
}
2025-06-27T15:28:01-06:00 INFO  [lsp] Language server with id 10 sent unhandled notification eslint/status:
{
  "uri": "file:///home/mk/Desktop/CODE/amboss-auth/src/app/actions.ts",
  "state": 1,
  "validationTime": 33
}
2025-06-27T15:28:07-06:00 ERROR [lsp] Server reset connection for a request "shutdown" id 2
2025-06-27T15:28:07-06:00 ERROR [lsp] Server reset connection for a request "shutdown" id 6
2025-06-27T15:28:07-06:00 ERROR [lsp] Server reset connection for a request "shutdown" id 6
2025-06-27T15:28:07-06:00 ERROR [gpui] window not found
2025-06-27T15:28:07-06:00 ERROR [gpui] window not found
2025-06-27T15:39:35-06:00 INFO  [lsp] Language server with id 7 sent unhandled notification eslint/status:
{
  "uri": "file:///home/mk/Desktop/CODE/magma/src/app/actions.ts",
  "state": 2
}
2025-06-27T15:39:36-06:00 INFO  [lsp] Language server with id 7 sent unhandled notification eslint/status:
{
  "uri": "file:///home/mk/Desktop/CODE/magma/src/app/actions.ts",
  "state": 2
}
2025-06-27T15:40:40-06:00 WARN  [project.format.local.code-actions] No code actions were resolved, continuing
2025-06-27T15:45:59-06:00 INFO  [db] Opening database /home/mk/.local/share/zed/db/0-stable/db.sqlite
```

</details>

### Zed Version and System Specs

Zed: v0.192.7 (Zed) 
OS: Linux Wayland zorin 17
Memory: 31.2 GiB
Architecture: x86_64
GPU: Intel(R) Xe Graphics (TGL GT2) || Intel open-source Mesa driver || Mesa 23.2.1-1ubuntu3.1~22.04.3

<img width="568" alt="Image" src="https://github.com/user-attachments/assets/bd555a6d-73b7-4eac-8ad7-34e2dc9836da" />

shows me

<img width="869" alt="Image" src="https://github.com/user-attachments/assets/f7a680f1-73c1-462c-b521-005a8c91aa46" />

```
ESLint server running in node v22.12.0
ESLint library loaded from: /Users/someonetoignore/Downloads/zed-eslint-broken/node_modules/.pnpm/eslint@9.30.1_jiti@2.4.2/node_modules/eslint/lib/unsupported-api.js

Failed to load plugin 'react-hooks' declared in ' » eslint-config-next/core-web-vitals » /Users/someonetoignore/Downloads/zed-eslint-broken/node_modules/.pnpm/eslint-config-next@15.3.4_eslint@9.30.1_jiti@2.4.2__typescript@5.8.3/node_modules/eslint-config-next/index.js': Cannot find module 'eslint-plugin-react-hooks'
Require stack:
- /Users/someonetoignore/Downloads/zed-eslint-broken/__placeholder__.js
Referenced from: /Users/someonetoignore/Downloads/zed-eslint-broken/node_modules/.pnpm/eslint-config-next@15.3.4_eslint@9.30.1_jiti@2.4.2__typescript@5.8.3/node_modules/eslint-config-next/index.js
Happened while validating /Users/someonetoignore/Downloads/zed-eslint-broken/src/app/actions.ts
This can happen for a couple of reasons:
1. The plugin name is spelled incorrectly in an ESLint configuration file (e.g. .eslintrc).
2. If ESLint is installed globally, then make sure 'eslint-plugin-react-hooks' is installed globally as well.
3. If ESLint is installed locally, then 'eslint-plugin-react-hooks' isn't installed correctly.

Consider running eslint --debug /Users/someonetoignore/Downloads/zed-eslint-broken/src/app/actions.ts from a terminal to obtain a trace about the configuration files used.
```

I did `pnpm i` beforehand, so this looks odd.
Is Zed not loading extra eslint libraries?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -492,7 +492,7 @@ libc = "0.2"
 libsqlite3-sys = { version = "0.30.1", features = ["bundled"] }
 linkify = "0.10.0"
 log = { version = "0.4.16", features = ["kv_unstable_serde", "serde"] }
-lsp-types = { git = "https://github.com/zed-industries/lsp-types", rev = "c9c189f1c5dd53c624a419ce35bc77ad6a908d18" }
+lsp-types = { git = "https://github.com/zed-industries/lsp-types", rev = "6add7052b598ea1f40f7e8913622c3958b009b60" }
 markup5ever_rcdom = "0.3.0"
 metal = "0.29"
 moka = { version = "0.12.10", features = ["sync"] }

--- a/crates/languages/src/typescript.rs
+++ b/crates/languages/src/typescript.rs
@@ -340,7 +340,7 @@ impl TypeScriptContextProvider {
 
 async fn detect_package_manager(
     worktree_root: PathBuf,
-    fs: &dyn Fs,
+    fs: Arc<dyn Fs>,
     package_json_data: Option<PackageJsonData>,
 ) -> &'static str {
     if let Some(package_json_data) = package_json_data {
@@ -445,7 +445,7 @@ impl ContextProvider for TypeScriptContextProvider {
                 let package_json_data = task.await.log_err();
                 vars.insert(
                     TYPESCRIPT_RUNNER_VARIABLE,
-                    detect_package_manager(worktree_root, fs.as_ref(), package_json_data.clone())
+                    detect_package_manager(worktree_root, fs, package_json_data.clone())
                         .await
                         .to_owned(),
                 );
@@ -822,22 +822,18 @@ impl LspAdapter for EsLintLspAdapter {
 
     async fn workspace_configuration(
         self: Arc<Self>,
-        fs: &dyn Fs,
+        _: &dyn Fs,
         delegate: &Arc<dyn LspAdapterDelegate>,
         _: Arc<dyn LanguageToolchainStore>,
         cx: &mut AsyncApp,
     ) -> Result<Value> {
         let workspace_root = delegate.worktree_root_path();
-        let worktree_root = delegate.worktree_root_path().to_path_buf();
         let use_flat_config = Self::FLAT_CONFIG_FILE_NAMES
             .iter()
             .any(|file| workspace_root.join(file).is_file());
 
-        let package_manager = detect_package_manager(worktree_root, fs, None).await;
-
         let mut default_workspace_configuration = json!({
             "validate": "on",
-            "packageManager": package_manager,
             "rulesCustomizations": [],
             "run": "onType",
             "nodePath": null,

--- a/crates/languages/src/typescript.rs
+++ b/crates/languages/src/typescript.rs
@@ -340,7 +340,7 @@ impl TypeScriptContextProvider {
 
 async fn detect_package_manager(
     worktree_root: PathBuf,
-    fs: Arc<dyn Fs>,
+    fs: &dyn Fs,
     package_json_data: Option<PackageJsonData>,
 ) -> &'static str {
     if let Some(package_json_data) = package_json_data {
@@ -445,7 +445,7 @@ impl ContextProvider for TypeScriptContextProvider {
                 let package_json_data = task.await.log_err();
                 vars.insert(
                     TYPESCRIPT_RUNNER_VARIABLE,
-                    detect_package_manager(worktree_root, fs, package_json_data.clone())
+                    detect_package_manager(worktree_root, fs.as_ref(), package_json_data.clone())
                         .await
                         .to_owned(),
                 );
@@ -782,8 +782,8 @@ pub struct EsLintLspAdapter {
 }
 
 impl EsLintLspAdapter {
-    const CURRENT_VERSION: &'static str = "2.4.4";
-    const CURRENT_VERSION_TAG_NAME: &'static str = "release/2.4.4";
+    const CURRENT_VERSION: &'static str = "3.0.10";
+    const CURRENT_VERSION_TAG_NAME: &'static str = "release/3.0.10";
 
     #[cfg(not(windows))]
     const GITHUB_ASSET_KIND: AssetKind = AssetKind::TarGz;
@@ -822,18 +822,22 @@ impl LspAdapter for EsLintLspAdapter {
 
     async fn workspace_configuration(
         self: Arc<Self>,
-        _: &dyn Fs,
+        fs: &dyn Fs,
         delegate: &Arc<dyn LspAdapterDelegate>,
         _: Arc<dyn LanguageToolchainStore>,
         cx: &mut AsyncApp,
     ) -> Result<Value> {
         let workspace_root = delegate.worktree_root_path();
+        let worktree_root = delegate.worktree_root_path().to_path_buf();
         let use_flat_config = Self::FLAT_CONFIG_FILE_NAMES
             .iter()
             .any(|file| workspace_root.join(file).is_file());
 
+        let package_manager = detect_package_manager(worktree_root, fs, None).await;
+
         let mut default_workspace_configuration = json!({
             "validate": "on",
+            "packageManager": package_manager,
             "rulesCustomizations": [],
             "run": "onType",
             "nodePath": null,
@@ -864,6 +868,7 @@ impl LspAdapter for EsLintLspAdapter {
             "experimental": {
                 "useFlatConfig": use_flat_config,
             },
+            "useFlatConfig": use_flat_config,
         });
 
         let override_options = cx.update(|cx| {

--- a/crates/languages/src/typescript.rs
+++ b/crates/languages/src/typescript.rs
@@ -782,8 +782,8 @@ pub struct EsLintLspAdapter {
 }
 
 impl EsLintLspAdapter {
-    const CURRENT_VERSION: &'static str = "3.0.10";
-    const CURRENT_VERSION_TAG_NAME: &'static str = "release/3.0.10";
+    const CURRENT_VERSION: &'static str = "2.4.4";
+    const CURRENT_VERSION_TAG_NAME: &'static str = "release/2.4.4";
 
     #[cfg(not(windows))]
     const GITHUB_ASSET_KIND: AssetKind = AssetKind::TarGz;
@@ -867,8 +867,7 @@ impl LspAdapter for EsLintLspAdapter {
             },
             "experimental": {
                 "useFlatConfig": use_flat_config,
-            },
-            "useFlatConfig": use_flat_config,
+            }
         });
 
         let override_options = cx.update(|cx| {

--- a/crates/project/src/lsp_command.rs
+++ b/crates/project/src/lsp_command.rs
@@ -3822,7 +3822,7 @@ impl GetDocumentDiagnostics {
             code,
             code_description: match diagnostic.code_description {
                 Some(code_description) => Some(CodeDescription {
-                    href: lsp::Url::parse(&code_description).unwrap(),
+                    href: Some(lsp::Url::parse(&code_description).unwrap()),
                 }),
                 None => None,
             },
@@ -3898,7 +3898,7 @@ impl GetDocumentDiagnostics {
             tags,
             code_description: diagnostic
                 .code_description
-                .map(|desc| desc.href.to_string()),
+                .and_then(|desc| desc.href.map(|url| url.to_string())),
             message: diagnostic.message,
             data: diagnostic.data.as_ref().map(|data| data.to_string()),
         })

--- a/crates/project/src/lsp_store.rs
+++ b/crates/project/src/lsp_store.rs
@@ -9130,7 +9130,13 @@ impl LspStore {
             }
         };
 
-        let lsp::ProgressParamsValue::WorkDone(progress) = progress.value;
+        let progress = match progress.value {
+            lsp::ProgressParamsValue::WorkDone(progress) => progress,
+            lsp::ProgressParamsValue::WorkspaceDiagnostic(_) => {
+                return;
+            }
+        };
+
         let language_server_status =
             if let Some(status) = self.language_server_statuses.get_mut(&language_server_id) {
                 status
@@ -10503,7 +10509,7 @@ impl LspStore {
                         code_description: diagnostic
                             .code_description
                             .as_ref()
-                            .map(|d| d.href.clone()),
+                            .and_then(|d| d.href.clone()),
                         severity: diagnostic.severity.unwrap_or(DiagnosticSeverity::ERROR),
                         markdown: adapter.as_ref().and_then(|adapter| {
                             adapter.diagnostic_message_to_markdown(&diagnostic.message)
@@ -10530,7 +10536,7 @@ impl LspStore {
                                     code_description: diagnostic
                                         .code_description
                                         .as_ref()
-                                        .map(|c| c.href.clone()),
+                                        .and_then(|d| d.href.clone()),
                                     severity: DiagnosticSeverity::INFORMATION,
                                     markdown: adapter.as_ref().and_then(|adapter| {
                                         adapter.diagnostic_message_to_markdown(&info.message)

